### PR TITLE
Fetch fields from Google Sheet

### PR DESCRIPTION
**Pull request by gidsg**
* Introduces lib `sheettodom` which fetches fields from a google sheet and populates data in the DOM based on the corresponding `data-sheet-attribute` value
* Refactor to build *shadow DOM* and then swap out loading div only when AJAX calls have completed. This tightly couples the AJAX requests with the interactive, but does reduce 'jank' of DOM mutations.

CC @akash1810

### CODE REVIEW

File: src/js/lib/sheettodom.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    var sheet = sheetUrl(sheetID);
+
+    reqwest({
+        'url': sheet,
+        'type': 'json',
+        'crossOrigin': true,
+        'success': resp => {
+
+            //get list of elements with data-sheet-attribute
+            for (const node of el.querySelectorAll('[data-sheet-attribute]')) {
```
Comment (akash1810) [inline src/js/lib/sheettodom.js]: minor: why not `el.querySelectorAll('[data-sheet-attribute]').forEach()`?

File: src/js/lib/sheettodom.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                const value = node.getAttribute('data-sheet-attribute');
+                node.innerHTML = resp.sheets[sheetName][0][value];
+            }
+            callback()
```
Comment (akash1810) [inline src/js/lib/sheettodom.js]: missing `;`

File: src/js/main.js [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
 export function init(el, context, config, mediator) {
-    el.innerHTML = mainHTML.replace(/%assetPath%/g, config.assetPath);
+const builder = document.createElement('div');
+builder.innerHTML = mainHTML.replace(/%assetPath%/g, config.assetPath);
 
-    [].slice.apply(el.querySelectorAll('.interactive-share')).forEach(shareEl => {
-        var network = shareEl.getAttribute('data-network');
-        shareEl.addEventListener('click', () => shareFn(network));
-    });
+    sheetToDOM('1NcSvccw77rHAncarwfeq7RGZF7yez9mP1Icf3oLMA7g', config.sheetName, builder, function callback(){
```
Comment (akash1810) [inline src/js/main.js]: should the sheet id be part of the config object?

File: src/js/main.js [hunk 4]
```diff
@@ -1,41 +1,47 @@
...
+        [].slice.apply(builder.querySelectorAll('.interactive-share')).forEach(shareEl => {
+            var network = shareEl.getAttribute('data-network');
+            shareEl.addEventListener('click', () => shareFn(network));
+        });
 
-    pimpYouTubePlayer('-Gy7poRbUHY', 'ytGuPlayer', '100%', '100%');
-    getYouTubeVideoDuration('-Gy7poRbUHY', function(duration) {
-      el.querySelector('.docs__poster--play-button').setAttribute("data-duration", duration);
-    })
+        getYouTubeVideoDuration('-Gy7poRbUHY', function(duration) {
```
Comment (akash1810) [inline src/js/main.js]: could the video id be read from the sheet?

File: src/js/lib/sheettodom.js (hunk 1 above)
Comment (gidsg) [inline src/js/lib/sheettodom.js]: Unfortunately this does not work on a nodelist. See https://developer.mozilla.org/en/docs/Web/API/NodeList#Why_is_NodeList_not_an_Array 
I got this style of iterator from: https://jakearchibald.com/2014/iterators-gonna-iterate/

File: src/js/main.js (hunk 4 above)
Comment (gidsg) [inline src/js/main.js]: Yes, that is absolutely an option, if we are happy for editorial to populate it.

**Comment by akash1810** [comment]
Strange mix of ES5 and ES6, but other than that 👍

File: src/js/main.js (hunk 3 above)
Comment (gidsg) [inline src/js/main.js]: Yes, it could be, though it is unlikely to change, I think it does make more sense in the config object.

**Comment by gidsg** [comment]
Thanks for the feedback @akash1810 I will merge this now and look at pulling more stuff into the sheet and config in a subsequent PR.

**Pull request merged**